.github/actions/integration-tests/run-ci-stage1

```diff
@@ -269,8 +269,31 @@ function run_and_capture_cmd {
 
 function post_run_cmd {
     run_and_capture_cmd "crucible extract run-id /var/lib/crucible/run/latest"
+    local run_id="${captured_output}"
 
-    run_cmd "crucible rm --run ${captured_output}"
+    run_and_capture_cmd "crucible extract primary-periods /var/lib/crucible/run/latest"
+    local primary_periods="${captured_output}"
+
+    if [ ${RC_STATUS} == 0 -a -n "${primary_periods}" ]; then
+        tmp_file=$(mktemp)
+        echo -e "${primary_periods}" > ${tmp_file}
+
+        OLD_IFS=${IFS}
+        while IFS=$'\n' read -r -u 9 line; do
```

Comment:
I was wondering how the file descriptor 9 fixed the issue on this loop

Reply:
I haven't completely wrapped my head around it myself, but I suppose what is happening is that since STDIN is "shared" with child processes that any manipulation/changes that the child process makes to it can somehow float back upstream.  In this case our buffer probably gets cleared and so on the next iteration through the loop there is nothing to read anymore and the read fails.

Reply:
Oh, and so moving it to a non-standard file handle avoids that interaction problem.

Reply:
I suppose the bigger takeaway is that we should probably never do anything involving STDIN being piped to something (ie. either a loop processor as is done here and/or a backgrounded task) while simultaneously forking off a new process.  The newly forked off process will acquire STDIN and could do something with it -- in the best case this just flushes the buffer but in the worst case could lead to the forked process doing something unintended using the input it receives from STDIN.

Reply:
cool thanks for the clarifications